### PR TITLE
WIP on new menu spec

### PR DESCRIPTION
**Pull request by wodenx**
Contains suggestions/specifications for bodiless 1.0 package refactors

### CODE REVIEW

File: examples/test-site/src/components/Menu/Refactor.md [hunk 1]
```diff
@@ -0,0 +1,398 @@
+# Navigation and Breadcrumbs refactor
+
+## Goal
+Improved DX and reduced LOE for activating navigation.
+
+## High Level
```
Comment (beliayeu) [inline examples/test-site/src/components/Menu/Refactor.md:6]: do we want to make all changes as backward-compatible? or are we fine with introducing breaking changes?

Comment (wodenx) [reply examples/test-site/src/components/Menu/Refactor.md:6]: breaking changes are ok

File: examples/test-site/src/components/Menu/Refactor.md [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+
+### First a token to apply generic top nav styles exported from @bodiless/navigation
+
+```js
+const $asTopNav = flow(
+  asTopNav,
+  // withSubMenuDesign is new helper which lets you target submenus directly
+  withSubMenuDesign({
+    List: asTopNavList,
+    Touts: asTopNavTouts,
```
[the remark this answers is not in the document]
Comment (wodenx) [reply examples/test-site/src/components/Menu/Refactor.md:63]: Not sure.
```
  withToutSubMenuDesign = (design) => withSubMenuDesign({
    Touts: withDesign(design),
    })
   ```
Feels fairly obvious to me.

File: examples/test-site/src/components/Menu/Refactor.md [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+);
+```
+
+## Apply styling
+
+### First a token to apply generic top nav styles exported from @bodiless/navigation
+
+```js
+const $asTopNav = flow(
+  asTopNav,
```
[the remark this answers is not in the document]
Comment (wodenx) [reply examples/test-site/src/components/Menu/Refactor.md:59]: yes - we already have it.

**Pull request closed without merging**